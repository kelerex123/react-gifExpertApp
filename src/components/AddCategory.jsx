import { useState } from "react"

const defaultFunction = () => {
    console.error("Se debe mandar obligatoriamente una funcion");
}

// export const AddCategory = ({setCategories}) => {
export const AddCategory = ({onNewCategory = defaultFunction}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();

        if(inputValue.trim().length <= 1) return;

        // setCategories(cat => [inputValue, ...cat]);
        onNewCategory(inputValue.trim());
        
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input 
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}