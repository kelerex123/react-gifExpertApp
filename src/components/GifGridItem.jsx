

export const GifGridItem = ({title="tituloDefecto", url="https://www.urlDefecto.com"}) => {
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}


