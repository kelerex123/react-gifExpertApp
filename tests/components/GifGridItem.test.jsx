import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem"

describe('Pruebas en GifGridItem.jsx', () => {
    
    const title = "One punch man";
    const url = "https://www.google.com/"

    test('Deberia hacer match con el snapshot ', () => {

        const {container} = render(<GifGridItem title={title} url={url}/>);
        
        expect(container).toMatchSnapshot();

    });

    test('Deberia mostrar la imagen con el URL y ALT por defecto', () => {

        render(<GifGridItem />);
        
        const {alt, src} = screen.getByRole('img');

        // expect(screen.getByRole('img').alt).toBe('tituloDefecto');
        // expect(screen.getByRole('img').src).toBe('https://www.urldefecto.com/');

        expect(alt).toBe('tituloDefecto');
        expect(src).toBe('https://www.urldefecto.com/');

    })

    test('Deberia mostrar la imagen con el URL y ALT indicado', () => {

        render(<GifGridItem title={title} url={url}/>);

        const {alt, src} = screen.getByRole('img');

        expect(alt).toBe(title);
        expect(src).toBe(url);

    })

    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifGridItem title={title} url={url}/>);

        expect(screen.getByText(title)).toBeTruthy();
    })
    
    

})
