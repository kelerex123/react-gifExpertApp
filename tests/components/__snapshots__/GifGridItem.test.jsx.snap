// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem.jsx Deberia hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One punch man"
      src="https://www.google.com/"
    />
    <p>
      One punch man
    </p>
  </div>
</div>
`;
